src/Development/IDE/Core/Compile.hs

```diff
@@ -69,9 +69,10 @@ parseModule
     -> Maybe SB.StringBuffer
     -> IO ([FileDiagnostic], Maybe ParsedModule)
 parseModule IdeOptions{..} env file =
+    let env' = env { hsc_dflags = hsc_dflags env `gopt_set` Opt_Haddock } in
```

Comment:
GHC has both `Opt_KeepRawTokenStream` and `Opt_Haddock` which are mutually incompatible. We need `Opt_KeepRawTokenStream` in DAML since HLint will otherwise miss some hints and we’ll need the same in `ghcide` if we start integrating `HLint`.

So hardcoding this to `Opt_Haddock` isn’t really an option. Sadly, I don’t have a great answer for what the right option would be. I guess for now, we could add a `ForceHadock` field to `IdeOptions` and then set this to `True` for `ghcide` and if we do integrate `HLint` we accept that we miss some hints since docs are probably more useful. Long-term this should really get fixed in GHC so that the two are not exclusive (iirc the long-term plan is to build haddock on top of `Opt_KeepRawTokenStream`).

Reply:
Yeah, that seemed too easy to not have been tried earlier.